architecture/README.md

```diff
@@ -122,49 +122,106 @@ class SomeViewController : UIViewController {
+
+### Usage
+
+The easiest way of creating an observable by calling the function `observableOf` or `subjectOf`:
 
 ```kotlin
-class SomeViewModel : BaseViewModel() {
-    private val flow = flowOf(1, 2, 3)
-    val flowObservable = flow.toObservable(coroutineScope)
+val observable = observableOf(1)
+```
+This will result in a `InitializedObservable<Int>`. Now it's possible to observe this value in several ways:
 
-    val flowable: BaseFlowable<Int> = SomeStateRepo() // someFlowable
-    val flowableSubject = flowable.toSubject(coroutineScope)
+```kotlin
 
-    fun readValue(defaultValue: Int): Int? {
-        return flowObservable.currentOrNull
-    }
+// property
+assertEquals(1, observable.current)
 
-    fun postValue(value: Int) {
-        flowableSubject.post(value)
-    }
+// observation
+observable.observe {
+    assertEquals(1, it)
 }
-```
 
-On the platform level observables can be observed. The platform specific observer will be notified of any changes to the observable.
-Observables are stateful, so any new observer will receive the last emitted value.
+// stateFlow
+observable.stateFlow.collect {
+    assertEquals(1, it)
+} 
 
-### Android
-On Android both observable and subject can easily be converted into `LiveData` objects (subject being `MutableLiveData`), allowing lifecycle-aware binding.
+// delegation 
+val oo:ObservableOptional.Value by observable
+assertEquals(1, oo.value)
+
+// delegation of the value is also possible 
+val i:Int by observable.valueDelegate
+assertEquals(1, i)
+
+```
+
+However since there is nothing backing the observable (other than an initial value), the value will never change. It makes more sense to make a subject this way:
 
 ```kotlin
-val observable: Observable<Int>
-val liveData = observable.liveData
 
-val subject: Subject<Int>
-val mutableLiveData = subject.liveData
+val subject = subjectOf(1)
 
-init {
-    liveData.observe(lifeCycle, Observer { value ->
-        mutableLiveData.postValue(value)
-    })
-}
+subject.post(2)
+assertEquals(2, observable.current)
+
+// or use delegation
+var i:Int by subject.valueDelegate
+assertEquals(2, i)
+i = 3
+assertEquals(3, observable.current)
 ```
 
+Another method is using the extension methods provided on many types of classes, `as*Observable()` or `as*Subject()`.
+
+For example on a `StateFlow`:
+```kotlin
+val stateFlow = MutableStateFlow<String>(null)
+val subject = stateFlow.toDefaultSubject()
+```
+
+Other than `(Mutable)StateFlow` there are extension methods for `Flow`, `(Mutable)SharedFlow`, `ReadOnlyProperty`, `ReadWriteProperty`.
+Be aware that properties can not be observed, only when a value is read from the observable is the backing `Property` checked for changes (which will then propagate to observers).
+
+### Android
+On Android both observable and subject can easily be converted into `LiveData` objects using the extension property `liveData`, allowing lifecycle-aware one way binding.
```

Comment:
I dont see LiveData anywhere in this code?

Reply:
See the android PR in the description.